tests/common/plugins/psu_controller/snmp_psu_controllers.py

```diff
@@ -171,6 +171,147 @@ def close(self):
         pass
 
 
+class APCSnmpPSU(PsuControllerBase):
+    """
+    PSU Controller class for 'APC Web/SNMP Management PSU'
+
+    This class implements the interface defined in PsuControllerBase class for PDU type 'APC Web/SNMP Management PSU'
+    """
+    PORT_NAME_BASE_OID = ".1.3.6.1.4.1.318.1.1.4.4.2.1.4"
+    PORT_STATUS_BASE_OID = ".1.3.6.1.4.1.318.1.1.12.3.5.1.1.4"
+    PORT_CONTROL_BASE_OID = ".1.3.6.1.4.1.318.1.1.12.3.3.1.1.4"
+    STATUS_ON = "1"
+    STATUS_OFF = "0"
+    CONTROL_ON = "1"
+    CONTROL_OFF = "2"
+
+    def _get_pdu_ports(self):
+        """
+        @summary: Helper method for getting PDU ports connected to PSUs of DUT
+
+        The PDU ports connected to DUT must have hostname of DUT configured in port name/description.
+        This method depends on this configuration to find out the PDU ports connected to PSUs of specific DUT.
+        """
+        try:
+            cmd = "snmpwalk -v 1 -c public -Ofenq %s %s " % (self.controller, self.PORT_NAME_BASE_OID)
```

Comment:
better to use pysnmp instead of calling command line.

```
    from pysnmp.proto import rfc1902
    from pysnmp.entity.rfc3413.oneliner import cmdgen
    from pyasn1.type import univ
```

example here.

https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/snmp_facts.py

Reply:
Thanks Guohan! I will update it with pysnmp.